### ISSUE TITLE
Can't get GP takeout to restore albums

### ISSUE THREAD

**Issue by CRCinAU** [simulot/immich-go#428]
I've been trying multiple imports to try and get albums included in the import to immich.

So far, no matter what I do, I can't seem to get this working... My latest command line I'm trying is:
```
./immich-go -server=X -key=Y upload -create-stacks=true -google-photos -create-albums=true -keep-untitled-albums -log-file=logfile.log -use-album-folder-as-name takeout-20240807T114245Z-001.zip
```

I also tried:
```
./immich-go -server=X -key=Y upload -create-stacks=true -google-photos -create-albums=true -keep-untitled-albums -log-file=logfile.log takeout-20240807T114245Z-001.zip
```

Using immich-go v0.21.1 and immich v1.111.0

I tried doing a second takeout export, and made sure to check all of the albums as well. Still doesn't seem to be create any albums in Immich.

Is there anything I should know or be trying with this?

ie, is this a side effect of using a Storage Template? Something else?

EDIT: After running through a second time - with the second command line above, stats show:
![image](https://github.com/user-attachments/assets/36857a2b-86d0-4cd2-9621-d6e8095cde95)

Still no albums :(

**Comment by simulot** [comment]
Are you sure that the takeout contains the albums?

**Comment by CRCinAU** [comment]
How can I tell? There doesn't seem to be many options during the takeout process.

I used:
![image](https://github.com/user-attachments/assets/c7f15a11-a427-4cdd-951e-e5a189b29ade)

Then the following:
![image](https://github.com/user-attachments/assets/d37944bd-c816-423f-914e-33ee1f4a0734)

There doesn't seem to be any further options to change / select.

**Comment by simulot** [comment]
It should be OK
Open the zip file, and check folders. Albums are placed in distinct folders.

Beware, google won't give you photos shared with you

**Comment by CRCinAU** [comment]
Hmmmm interesting.... The ZIP file just looks like:
![image](https://github.com/user-attachments/assets/221d0c4d-c2c1-4f7e-92cb-cb5d2afe06dd)

**Comment by simulot** [comment]
There is a curious "Locked Folder"... but no albums here

**Comment by CRCinAU** [comment]
The 'Locked Folder' just contains stuff that was in the private folder within Google Photos... Each folder just contains pictures and json metadata - I think its called the sidecar data?

Given the complete lack of options to change for the export, what is a guy to do in this scenario?

I've done a third export now and its in the middle of downloading again - just to see if theres anything different this time - but if its still the same, I'm not sure what to do...

**Comment by simulot** [comment]
You can click on the "All photos album included" to get the list of albums.
Remember that shared albums contain only your own photos.

**Comment by CRCinAU** [comment]
Yeah - I think that might be the problem - the Albums that *I* shared with others - which is most of them aren't included...

I'm going through and removing all the sharing from everything - and hopefully, recreating some of the albums I care about and see if this fixes the problem.

**Comment by simulot** [comment]
It won't change anything if you don't own the photos. In that case, you'll end up in downloading the album from the googles photos web page. But you'll get low quality pictures and no metadata

**Comment by CRCinAU** [comment]
Yeah - I own all these photos - its just that I shared them with others....

I'm doing some sorting in Google Photos first - because it doesn't look like I can then create an album from within Immich and then add all photos from a group of locations...

I'm guessing this is probably the issue - as it seems like after I unshared the other albums, they disappeared from the Albums list.

I'll close this for now - because I really think that's the problem. As you mentioned, if the ZIP file doesn't include any Albums, then theres no way that they can be imported.
